src/Lecture3.hs

```diff
@@ -125,9 +134,11 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  Reward rxg rxp <> Reward ryg ryp =
+    Reward (rxg <> ryg) (rxp || ryp)
 
 instance Monoid Reward where
+  mempty = Reward (mempty :: Gold) False
```

Comment:
You don't need to specify the type here explicitly as GHC will figure out it's `Gold` due to type inference 🥇 
```suggestion
  mempty = Reward mempty False
```